src/raw_client.rs

```diff
@@ -80,6 +81,16 @@ impl<'a> RawClient<'a> for RawClientImpl<'a> {
             .await
             .context(RequestTimeout {})?;
         let reply = result.context(ReadReply {})?;
+        if reply.get_request_id() != request.get_request_id() {
+            client_connection.connection.invalidate();
+        }
+        ensure!(
+            reply.get_request_id() == request.get_request_id(),
```

Comment:
You can just add a return call inside the if above that way you don't have to repeat the condition.